packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.error.js

```diff
@@ -1,150 +1,258 @@
-import algoliaClient from 'algoliasearch/lite';
 import createInstantSearchManager from '../createInstantSearchManager';
 
-jest.useFakeTimers();
-
-jest.mock('algoliasearch-helper/src/algoliasearch.helper.js', () => {
-  let count = 0;
-  const Helper = require.requireActual(
-    'algoliasearch-helper/src/algoliasearch.helper.js'
-  );
-  Helper.prototype._dispatchAlgoliaResponse = function(state) {
-    if (count > 3) {
-      this.emit('result', { count: count++ }, state);
-    } else {
-      this.emit('error', { count: count++ }, state);
-    }
-  };
-  Helper.prototype.searchForFacetValues = () =>
-    Promise.reject(new Error('error'));
-  return Helper;
+const createSearchClient = () => ({
+  search: jest.fn(() =>
+    Promise.resolve({
```

Comment:
maybe you can simplify this to just jest.fn(),
it seems you're always calling mockImplementation, which means it's unused?